virtual_rainforest/models/soil/env_factors.py

```diff
@@ -0,0 +1,306 @@
+        minimum_pH=constants.min_pH_microbes,
+        lower_optimum_pH=constants.lowest_optimal_pH_microbes,
+        upper_optimum_pH=constants.highest_optimal_pH_microbes,
+    )
+    clay_factor_saturation = calculate_clay_impact_on_enzyme_saturation(
+        clay_fraction=clay_fraction,
+        base_protection=constants.base_soil_protection,
+        protection_with_clay=constants.soil_protection_with_clay,
+    )
+    clay_factor_decay = calculate_clay_impact_on_necromass_decay(
+        clay_fraction=clay_fraction,
+        decay_exponent=constants.clay_necromass_decay_exponent,
+    )
+
+    # Combine all factors into a single EnvironmentalFactors object
+    return EnvironmentalFactors(
+        water=water_factor,
+        pH=pH_factor,
+        clay_saturation=clay_factor_saturation,
+        clay_decay=clay_factor_decay,
+    )
+
+
+def calculate_temperature_effect_on_microbes(
+    soil_temperature: NDArray[np.float32],
+    activation_energy: float,
+    reference_temperature: float,
+) -> NDArray[np.float32]:
+    """Calculate the effect that temperature has on microbial metabolic rates.
+
+    This uses a standard Arrhenius equation to calculate the impact of temperature.
+
+    This function takes temperatures in Celsius as inputs and converts them into Kelvin
+    for use in the Arrhenius equation. TODO - review this after we have decided how to
+    handle these conversions in general.
+
+    Args:
+        soil_temperature: The temperature of the soil [C]
+        activation_energy: Energy of activation [J mol^-1]
+        soil_temperature: The reference temperature of the Arrhenius equation [C]
+
+    Returns:
+        A multiplicative factor capturing the effect of temperature on microbial rates
+    """
+
+    # Convert the temperatures to Kelvin
+    soil_temp_in_kelvin = convert_temperature(
+        soil_temperature, old_scale="Celsius", new_scale="Kelvin"
+    )
+    ref_temp_in_kelvin = convert_temperature(
+        reference_temperature, old_scale="Celsius", new_scale="Kelvin"
+    )
+
+    return np.exp(
+        (-activation_energy / gas_constant)
+        * ((1 / (soil_temp_in_kelvin)) - (1 / (ref_temp_in_kelvin)))
+    )
+
+
+def calculate_water_potential_impact_on_microbes(
+    water_potential: NDArray[np.float32],
+    water_potential_halt: float,
+    water_potential_opt: float,
+    moisture_response_curvature: float,
+) -> NDArray[np.float32]:
+    """Calculate the effect that soil water potential has on microbial rates.
+
+    This function only returns valid output for soil water potentials that are less than
+    the optimal water potential.
+
+    Args:
+        water_potential: Soil water potential [kPa]
+        water_potential_halt: Water potential at which all microbial activity stops
+            [kPa]
+        water_potential_opt: Optimal water potential for microbial activity [kPa]
+        moisture_response_curvature: Parameter controlling the curvature of the moisture
```

Comment:
This is also a parameter that we both use, so we should coordinate where this is stored

Reply:
I'll rename this parameter as it _is_ a different parameter from the one you use, the names just overlap a bit making it easy to confuse them